### ISSUE TITLE
(Suggestion) Allow icon-only mode for side panel

### ISSUE THREAD

**Issue by alejandro-perez** [gmc-holle/xfdashboard#89]
As you have explained several times, minimized windows are a problem. The same happens with windows in other virtual desktops. You gracefully solve this issue by saving a snapshot of each window, I think this is a good idea. But sometimes you have both snapshots and icons on the side panel. While generally I'm interested on having previous of the windows of my current desktop, it would be enough for me to have icons for those windows in other desktops. Moreover, I think it might be clearer to me since I don't have two different kinds of information (either snapshot or icon). 

Since I guess this would be an easy change, could you add it as an option?

**Comment by gmc-holle** [comment]
Do I understand right that you would an option to tell xfdashboard to use either a snapshot of a window going to be minimized or to always use its window icon instead (like it does when a window minimized at start-up)?

**Comment by alejandro-perez** [comment]
The latter option. To use in the small workspaces the window icon instead. With two porpuses:

1) Sometimes two big white windows (e.g. chromium, text editor...) show very similar, and it's difficult to say which one is. 
2) Get some privacy at work. When I switch windows, sometimes my boss does not care whether I have facebook opened in the 5th workspace :P

Another alternative for 1) would be including a small window icon in addition to the snapshot.

**Comment by gmc-holle** [comment]
Regarding alternative to 1.) I don't think it would really help because the window icons may be overlapped by another window in workspace hiding the window icons. Think of three windows: Two big white one and a smaller black one. If the smaller black one is placed smart in workspace it would overlap both window icons of the big white window. At the end you have again to guess which big white one is which one.

But it should be possible to add a stylable (by CSS) property, e.g. 'show-window-content', to tell the window in workspace to show its window content or the window icon. In addition to the already pseudo-class ':active' which is only set at the current active workspace in the workspace selector the following CSS code would show the window content at the active workspace and window icon at the inactive ones:

```
XfdashboardLiveWorkspace
{
    show-window-content: false;
}

XfdashboardLiveWorkspace:active
{
    show-window-content: true;
}
```

I think a disadvantage could be that it will now take longer to display the windows when the active workspace changes. All windows have to be "loaded" and "setup" again because the actor will hold an active reference to the corresponding XfdashboardWindowContent instance any longer. That's the price for privacy :P

**Comment by alejandro-perez** [comment]
You're right. Leave it as it is, as it will probably not help much :).

Regards!

**Comment by gmc-holle** [comment]
Too late. It's implemented with commit 081f058f8251760bf2e07d71a0f3eb7a99753b33 ;)

**Comment by alejandro-perez** [comment]
Well, I guess I have to test it :)

**Comment by alejandro-perez** [comment]
Well, it works. I didn't get you, and I was expecting something different. But I kind of like it :). I will use it for privacy reasons :).

**Comment by alejandro-perez** [comment]
Sorry, but I don't think I like how they look in this mode. I'd rather see them as they look in the "normal" mode, when windows are unmaped (i.e. opaque rectangle wiith centered icon). Would that be possible? Even though they "might" get shadowed by other windows.

Regards!

**Comment by gmc-holle** [comment]
You want the windows at the "small" workspace to be themable in background color and window icon position? Ok, that should be possible. Give me some time.

**Comment by alejandro-perez** [comment]
Actually, I wanted to make them look like this (mouse position):
![captura de pantalla_2015-05-01_18-33-11](https://cloud.githubusercontent.com/assets/2082536/7433347/9c9b9df0-f030-11e4-9cab-f3b7b511eb3b.png)

**Comment by gmc-holle** [comment]
With commit 729bbb9ff2e012ecf995691bb7bd9e510cae0e68 I create a new fully themable actor called XfdashboardLiveWorkspaceWindow used in XfdashboardLiveWorkspace representing a window at the workspace.

I think you would like the window icon to fill to largest possible area of the window in the "small" workspace but also to keep the aspect ratio.

So replace in xfdashboard.css the code block

```
XfdashboardLiveWorkspace
{
    show-window-content: false;
}

XfdashboardLiveWorkspace:active
{
    show-window-content: true;
}
```

with

```
/* Live workspaces */
XfdashboardLiveWorkspace
{
	background-type: outline;
	background-image-type: desktop;
	outline-color: #404040ff;
	outline-width: 1.0;
}

XfdashboardLiveWorkspace:active
{
	background-type: outline;
	outline-color: #ffff00ff;
	outline-width: 2.0;
}

XfdashboardLiveWorkspace XfdashboardLiveWorkspaceWindow
{
	show-window-content: false;
	background-type: outline fill;
	background-fill-color: black;
	outline-color: white;
	outline-width: 1.0;
	window-icon-x-fill: true;
	window-icon-y-fill: true;
	window-icon-fill-keep-aspect: true;
}

XfdashboardLiveWorkspace:active XfdashboardLiveWorkspaceWindow
{
	show-window-content: true;
	background-type: none;
}
```

Is it ok?

**Comment by alejandro-perez** [comment]
It is absolutely perfect, thanks! I like this way much better!

**Comment by gmc-holle** [comment]
Reopen this issue because it causes problems as described in #44

**Comment by gmc-holle** [comment]
Hmmm ... I missed to tell that this feature was re-implemented with commit e5a615da7c2304f01b3716be62127cb66c85fcdc in January. The workspaces at the side panel now use the same actor base type (XfdashboardLiveWindowSimple) as the windows view. So the windows shown in workspace can be themed the same way as in windows view and XfdashboardLiveWindowSimple supports to show either the live content or the window icon.

So just replace the following CSS:
    
```
XfdashboardLiveWorkspace
{
    show-window-content: <true or false>;
}
```    
with
```
XfdashboardLiveWorkspace XfdashboardLiveWindowSimple
{
    display-type: <icon or live-preview>;
}
```
or add the latter.

I'm closing this issue now.
